packages/flame/lib/src/components/mixins/has_paint.dart

```diff
@@ -0,0 +1,74 @@
+import 'dart:ui';
+
+import '../../../components.dart';
+import '../../palette.dart';
+
+/// Adds a collection of paints to a component
+///
+/// Component will always have a main Paint that can be accessed
+/// by the [paint] attribute and other paints can be manipulated/accessed
+/// using [getPaint], [setPaint] and [deletePaint]
+mixin HasPaint<T extends Object> on BaseComponent {
```

Comment:
`T extends Object` === `T`? everything extends Object right?

Reply:
ahh sorry I see the difference in dart. so I was thinking we could have a check:

```
if (T == Object)
```

and forbid using the `_paints` map in this case. that way people are forced to either use HasPaint w/o any key type, which will fallback to Obj and not allow using the map, or provide a specific key type and "enable" the map.

Reply:
this assert method could be called on every method that provides a non-null `key/paintId`

Reply:
eg (dartpad):
```dart
void main() {
  A().foo();
  A<String>().foo();
}

class A<T extends Object> {
  void foo() {
    if (T != Object) {
      print('faaa $T');
    }
  }
}
```

Reply:
Sorry, I didn't understood if this is just your train of thought, or if there is a suggestion here.

Reply:
yes, I think we should have a check

Reply:
can you explain more what this check would be? I am lost here

Reply:
I think we should add a check so that we enforce one of two cases:
* use specifies the generic type T and thus can use the map
* user does not and thus T fallsback to Object, in which case we block using the map
Otherwise, everyone that doesn't specific T will be able to add any keys they want to components that should not have extra paints.